src/helpers.ts

```diff
@@ -198,3 +198,12 @@ interface SurvivalGameInfo {
      */
     wave: number;
 }
+
+interface _Constructor<T> {
+    readonly prototype: T;
```

Comment:
While this is done by `lib.es6.d.ts` and is somewhat commonplace, it has to be done with care to avoid associating instance-only (own enumerable properties of the object, in proto-OO terms) properties with the constructors prototype. Since many properties of screeps happen to be of this kind, it applies to a lot of the code here.

An example:

```typescript
interface StructureConstructor {
    readonly prototype: Structure;
}
declare const Structure: Structure;
```

Because  the `Structure` global gets the `prototype` field of the `Structure` interface, `Structure.prototype.id` is now typed as `number`, when in reality it is `undefined`, and should be a property-access error.

It's arguably rather subtle and will only happen with obscure use of prototypes though, so I can argue for just keeping this as-is. In fact, the standard library typings gets this wrong itself:

```typescript
interface RegExp {
    lastIndex: number;
}

interface RegExpConstructor {
    readonly prototype: RegExp;
}
```

Yet `RegExp.prototype.lastIndex` is `undefined`.